git/git.go

```diff
@@ -622,6 +622,63 @@ func IsVersionAtLeast(actualVersion, desiredVersion string) bool {
 	return actual >= atleast
 }
 
+// CloneWithoutFilters clones a git repo but without the smudge filter enabled
+// so that files in the working copy will be pointers and not real LFS data
+func CloneWithoutFilters(args []string) error {
+
+	// Disable the LFS filters while cloning to speed things up
+	// this is especially effective on Windows where even calling git-lfs at all
+	// with --skip-smudge is costly across many files in a checkout
+	cmdargs := []string{
+		"-c", "filter.lfs.smudge=",
+		"-c", "filter.lfs.required=false",
+		"clone"}
+	cmdargs = append(cmdargs, args...)
+	cmd := execCommand("git", cmdargs...)
+
+	// Spool stdout directly to our own
+	cmd.Stdout = os.Stdout
+	// stderr needs filtering
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("Failed to get stderr from git clone: %v", err)
+	}
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("Failed to start git clone: %v", err)
+	}
+
+	// Filter stderr to exclude messages caused by disabling the filters
+	// As of git 2.7 it still tries to call the blank filter but required=false
+	scanner := bufio.NewScanner(stderr)
+	for scanner.Scan() {
+		s := scanner.Text()
+		// Swallow all the known messages from intentionally breaking filter
+		if strings.Contains(s, "error: external filter") ||
+			strings.Contains(s, "error: cannot fork") ||
+			// Linux / Mac messages
+			strings.Contains(s, "error: cannot run : No such file or directory") ||
+			strings.Contains(s, "warning: Clone succeeded, but checkout failed") ||
+			strings.Contains(s, "You can inspect what was checked out with 'git status'") ||
+			strings.Contains(s, "retry the checkout") ||
```

Comment:
Hmm, maybe, I wasn't keen on assuming the number of lines surrounding an error. Maybe rather than trying to be super-precise about parsing sequences of error text (which might change after all), perhaps it's best just to suppress all the known errors like this in the output but to write them all to the git-lfs logs. If git clone fails it'll stop with a non-zero anyway and we could just suggest running 'git lfs logs last' to get the full story.